src/Interpreters/ClusterProxy/executeQuery.cpp

```diff
@@ -712,9 +712,9 @@ void executeQueryWithParallelReplicas(
     rewriteJoinToGlobalJoin(modified_query_tree, context);
     modified_query_tree = buildQueryTreeForShard(planner_context, modified_query_tree);
 
-    auto header
-        = InterpreterSelectQueryAnalyzer::getSampleBlock(modified_query_tree, context, SelectQueryOptions(processed_stage).analyze());
     auto modified_query_ast = queryNodeToDistributedSelectQuery(modified_query_tree);
+    auto header
+        = InterpreterSelectQueryAnalyzer::getSampleBlock(modified_query_ast, context, SelectQueryOptions(processed_stage).analyze());
```

Comment:
We should not do it. The whole point is to avoid conversation: Query Tree -> AST -> Query Tree